flexbox/src/main/java/com/google/android/flexbox/FlexboxLayout.java

```diff
@@ -614,64 +623,91 @@ private void determineMainSize(@FlexDirection int flexDirection, int widthMeasur
      */
     private int expandFlexItems(FlexLine flexLine, @FlexDirection int flexDirection,
             int maxMainSize, int paddingAlongMainAxis, int startIndex) {
+        int childIndex = startIndex;
```

Comment:
As per https://www.w3.org/TR/css-flexbox-1/#resolve-flexible-lengths min constraints are only checked after space is distributed. This could mean that an item was expanded but did not meet the min size. (Say it had a width of wrap content and happened to be less than the min width and it did not receive enough free space when expanding).